esp-client/src/main.cpp

```diff
@@ -109,49 +71,12 @@ void loop()
 		reconnect();
 	}
 
-	// FIXME: It does not get detected when no scale is connected
-	if (scale.is_ready()) // check if scale is ready
+	if (scale.weightChanged())
 	{
-		Serial.println("scale is ready!");
-
-		static bool weightChange;			  // saves if weight changed above or below threshold inbetween two readings
-		static bool printed;				  // saves wether settled weight has already been sent via MQTT and printed to the serial monitor
-		static float previousWeight = weight; // saves the value of the previous measurement
-		weight = readScale();
-
-		if (weight >= previousWeight + SCALE_THRESHOLD || weight <= previousWeight - SCALE_THRESHOLD)
-		{ // if weight changed above or below threshold
-			weightChange = true;
-			printed = false;
-
-			Serial.print("Change detected, weight: "); // print to serial monitor
-			Serial.print(weight, 1);
-			Serial.println("g");
-
-			previousWeight = weight;
-			weight = readScale();
+		float weight = scale.getCurrentWeight();
 
-			delay(250);
-		}
-		else
-		{
-			weightChange = false;
-		}
-
-		if (!weightChange && !printed)
-		{									// if weight changed over threshold
-			Serial.print("final weight: "); // print to serial monitor
-			Serial.print(weight);
-			Serial.println("g");
-
-			weightChange = false; // reset change
-
-			sendWeight(weight); // send weight over MQTT
-
-			printed = true;
-		}
```

Comment:
sollte nicht entfernt werden bzw. alternativ anders in `scale.cpp` implementiert werden, sonst viele MQTT-messages während sich die Waage noch einpendelt (z.B: `270g, 100g, 30g, 0g` wenn das Gewicht entfernt wird statt `270g, 0g`)

Reply:
So wie ich das rauslese, ist `printed` das gleiche wie `!weightChange`, dem entsprechend ist das auch rausgekürzt.